### PR TITLE
K8s: Improve identity mapping setup

### PR DESCRIPTION
**Pull request by ryantxu**
We currently hack access to *SignedInUser in a few places.  This PR flips some of the logic and makes the mappings more explicit at the beginning.

The two key cases are:
1. When running inside the grafana proxy, we now add the user to context at startup
2. The k8s handler chain now explicitly tries to map a k8s user to grafana requester

This is extracted from https://github.com/grafana/grafana/pull/89444 -- since that seems to be more complicated than expected 😢  

Most of the PR is moving `appcontext.User(ctx)` >> `identity.GetRequester(ctx)` -- this will help ensure we are testing this dependency and can move more things to the interface rather than struct implementation.

### CODE REVIEW

File: pkg/infra/appcontext/user.go [hunk 1]
```diff
@@ -38,44 +33,32 @@ func User(ctx context.Context) (*user.SignedInUser, error) {
 		return grpcCtx.SignedInUser, nil
 	}
 
-	// Set by incoming HTTP request
-	c, ok := ctxkey.Get(ctx).(*contextmodel.ReqContext)
```
Comment (ryantxu) [inline pkg/infra/appcontext/user.go:42]: ^^^ this was previously doing the identity mapping in the grafana proxy

File: pkg/services/apiserver/service.go [hunk 2]
```diff
@@ -157,6 +157,11 @@ func ProvideService(
 				req.URL.Path = "/"
 			}
 
+			if c.SignedInUser != nil {
+				ctx := appcontext.WithUser(req.Context(), c.SignedInUser)
+				req = req.WithContext(ctx)
+			}
```
Comment (ryantxu) [inline pkg/services/apiserver/service.go:163]: This is now how our proxy authentication is registered

File: pkg/apiserver/builder/helper.go [hunk 3]
```diff
@@ -96,6 +96,7 @@ func SetupConfig(
 
 		// Needs to run last in request chain to function as expected, hence we register it first.
 		handler := filters.WithTracingHTTPLoggingAttributes(requestHandler)
+		handler = filters.WithRequester(handler)
```
Comment (ryantxu) [inline pkg/apiserver/builder/helper.go:99]: ^^^ and we now map the `system:master` user here

**Review by charandas — APPROVED** [review]
LGTM! Thanks for the helpful comments.

**Review by Jguer — APPROVED** [review]
awesome change, thanks @ryantxu

File: pkg/infra/appcontext/user.go [hunk 4]
```diff
@@ -38,44 +37,32 @@ func User(ctx context.Context) (*user.SignedInUser, error) {
...
-							dashboards.ActionDashboardsDelete: {"*"},
-							dashboards.ActionFoldersCreate:    {"*"},
-							dashboards.ActionFoldersRead:      {dashboards.ScopeFoldersAll}, // access to read all folders
-						},
-					},
-				}, nil
-			}
-		}
+	// If the identity was set via requester, but not appcontext, we can map values
+	// NOTE: this path
```
Comment (DanCech) [inline pkg/infra/appcontext/user.go:41]: are we missing the rest of the comment here?

**Pull request merged**